TodoApp.ConsoleApp/UI/UserInput.cs

```diff
@@ -0,0 +1,164 @@
+    {
+        const string Yes = "yes";
+        const string No = "no";
+
+        private static string ReadInput(bool required = false, string defaultValue = "")
+        {
+            string text = "";
+            if (required)
+            {
+                text = Console.ReadLine();
+
+                while (string.IsNullOrEmpty(text))
+                {
+                    Console.WriteLine("This field can't be empty!");
+                    text = Console.ReadLine();
+                }
+            }
+
+            else
+            {
+                text = Console.ReadLine();
+
+                if (string.IsNullOrEmpty(text))
+                {
+                    text = defaultValue;
+                }
+            }
+
+            return text;
+        }
+
+        public static string ReadText(string question, bool required = false, string defaultValue = "")
+        {
+            Console.WriteLine(question);
+            string text = "";
+            if (required)
+            {
+                text = ReadInput(required);
+
+                if (string.IsNullOrEmpty(text))
+                {
+                    text = defaultValue;
+                }
+            }
+
+            else
+            {
+                text = defaultValue;
+            }
+
+            return text;
+        }
+
+        public static int ReadOption(string question, string[] availableOptions, bool required = false, int defaultValue = 0)
+        {
+            Console.WriteLine(question);
+            int index = 1;
+            foreach (var option in availableOptions)
+            {
+                Console.WriteLine($"\t{index}: {option}");
+                index++;
+            }
+
+            index = 1;
+            bool isValid = false;
+            string input = "";
+
+            if (required)
+            {
+                do
+                {
+                    input = ReadInput(required);
+                    isValid = int.TryParse(input, out index) && (index > 0 && index <= availableOptions.Length);
+
+                    if (!isValid)
+                    {
+                        Console.WriteLine($"Please, enter a valid number between 1 and {availableOptions.Length}!");
+                    }
+                }
+                while (!isValid);
+            }
+
+            else
+            {
+                input = ReadInput(required);
+                isValid = int.TryParse(input, out index) && (index > 0 && index < availableOptions.Length);
+                if (!isValid)
+                {
+                    index = defaultValue;
+                }
+            }
+
+            return index;
+        }
+
+        public static bool ReadYesNo(string question, bool required = false, bool defaultValue = false)
+        {
+            Console.WriteLine(question);
+            string choice = "";
+            bool isValid = false;
+            bool result = false;
+
+            if (required)
```

Comment:
Same as `ReadOptions()` - remove the `if-else`, keep the `do-while`, add parameters to `ReadInput()`.